### PR TITLE
update(platform): service disk autoscaler GA

### PR DESCRIPTION
**Pull request by wojcik-dorota**
https://aiven.atlassian.net/browse/AD-2180

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

### CODE REVIEW

**Review by ilpon — APPROVED** [review]

File: docs/platform/howto/disk-autoscaler.md [hunk 1]
```diff
@@ -57,20 +57,33 @@ the available disk storage capacity and depends on a service type:
 -   Aiven for Apache Cassandra®: 35% of the available disk storage
     capacity
 -   Aiven for OpenSearch®: 75% of the available disk storage capacity
--   All the other Aiven service types: 85% of the available disk storage
-    capacity
+-   All the other Aiven service types: 85% of the available disk storage capacity
```
Comment (caydinavn) [inline docs/platform/howto/disk-autoscaler.md:60]:
```suggestion
-   All other Aiven service types: 85% of the available disk storage capacity
```

File: docs/platform/howto/disk-autoscaler.md [hunk 2]
```diff
@@ -57,20 +57,33 @@ the available disk storage capacity and depends on a service type:
...
+  limited by:
+  - Maximum disk capacity set for the
+    [Aiven Autoscaler endpoint](#change-max-disk-space)
+  - Maximum [DDS](/docs/platform/howto/add-storage-space) storage capacity supported on
+    your service plan
+- When triggered, the autoscaling process takes a moment. Meanwhile, the service disk
+  might get full, and your service might enter the read-only mode. In such cases, the
+  service is back to normal as soon as the autoscaling process completes unless Aiven
+  Autoscaler's disk capacity limits are exceeded.
+- Disk autoscaling works on fully running services only and cannot happen during
```
Comment (caydinavn) [inline docs/platform/howto/disk-autoscaler.md:76]: We might be able to drop this limitation, worth checking with @ilpon

**Review by caydinavn — CHANGES_REQUESTED** [review]

File: docs/platform/howto/disk-autoscaler.md (hunk 2 above)
Comment (ilpon) [reply docs/platform/howto/disk-autoscaler.md:72]: It requires some work (to ignore old nodes) and I'd like to first see that DDS works when applied manually during maintenance operations before making autoscaler do it.

In other words, I'd leave it for future enhancement.

**Review by caydinavn — APPROVED** [review]

**Review by ArthurFlag — APPROVED** [review]

**Review by ilpon — CHANGES_REQUESTED** [review]
Based on discussion in ORR, requesting few changes.

First in the "How it works" section the "30 seconds" is incorrect. Maybe something like:

>  From that point onward, the disk space availability of your service is monitored by autoscaler.

To give understanding of what to expect without promising too much, we could add "usually within minutes" to the next step:

> When disk storage consumption reaches the threshold for a specific service, disk autoscaler increases available storage space usually within minutes by 10% every time taking the used disk space as a baseline.

Then we could add new step(s) to tell that after this the disk increase is recorded in project event log and a notification is sent about the added disk.

And lastly, it was requested to warn the user about the conflict between autoscaler and terraform in current terraform version. (Both are trying to control the disk space. As far as I know, nothing has been done for this and support for autoscaler in terraform is waiting for the GA.)

**Review by aiven-amartin — APPROVED** [review]
The changes to reflect the ORR concerns make sense to me, thanks!

File: docs/platform/howto/disk-autoscaler.md [hunk 3]
```diff
@@ -53,20 +53,38 @@ the available disk storage capacity and depends on a service type:
 -   Aiven for Apache Cassandra®: 35% of the available disk storage
     capacity
 -   Aiven for OpenSearch®: 75% of the available disk storage capacity
--   All the other Aiven service types: 85% of the available disk storage
-    capacity
+-   All other Aiven service types: 85% of the available disk storage capacity
+
+1.  The disk increase is recorded in the project event log, and you receive a notification
+    about the added disk space.
```
Comment (ilpon) [inline docs/platform/howto/disk-autoscaler.md:59]: This doesn't render correctly.

**Review by ilpon — CHANGES_REQUESTED** [review]

**Review by ilpon — APPROVED** [review]

**Pull request merged**